src/fluree/db/ledger/transact.clj

```diff
@@ -146,14 +147,11 @@
                                        (-> session :conn :group) (:network session)
                                        (:dbid session) (dissoc block-result :db-before :db-after)))]
               (if (true? new-block-resp)
-                (do
-                  ;; update cached db
-                  ;(let [new-db-ch (async/promise-chan)]
-                  ;  (async/put! new-db-ch (:db-after block-result))
-                  ;  (session/cas-db! session db-before-ch new-db-ch))
-                  ;; reindex if needed
-                  ;; to do -add opts
-                  (<? (indexing/index* session {:remove-preds remove-preds*}))
+                (let [res (<? (indexing/index* session {:remove-preds remove-preds*}))]
+                  (when-not res
+                    (let [new-db-ch (async/promise-chan)]
+                      (async/put! new-db-ch (:db-after block-result))
+                      (session/cas-db! session db-before-ch new-db-ch)))
```

Comment:
This code shouldn't be needed, was there an issue that was solved causing you to re-introduce this? The DB should get updated above as a result of txproto/propose-new-block-async - which blocks. This proposes the new DB to consensus, and if successful it will update the DB in the cache.

If there was an issue however we can look into fixing it, but right now this is redundant and might actually create issues.

Reply:
This was resolved with the update to the transactor/message code